homeassistant/components/rituals_perfume_genie/switch.py

```diff
@@ -89,4 +96,9 @@ async def async_turn_off(self, **kwargs):
 
     async def async_update(self):
         """Update the data of the device."""
-        await self._diffuser.update_data()
+        try:
+            await self._diffuser.update_data()
+            self._available = self._diffuser.data["hub"]["status"] == AVAILABLE_STATE
```

Comment:
Please only wrap the line that can raise in the try.. .except block. Use an `else:` block or return early in the except block if needed.